### PR TITLE
Fix deployed code publication for packages with unit tests

### LINKED ISSUE #505: Deployed code packaging fails with <ILLEGAL VALUE> in package with unit tests
See https://github.com/intersystems/ipm/blob/v1/src/cls/IPM/Lifecycle/Module.cls#L109 - need smarter handling for "no extension" and/or alternative extensions.

### PR DESCRIPTION
**Pull request by isc-tleavitt**
Fixes #505

### CODE REVIEW

File: tests/sandbox/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+ && apt-get install -y jq \
+ && apt-get install -y git
 
-USER irisowner
\ No newline at end of file
+USER irisowner
+
+RUN --mount=type=bind,src=.,dst=/home/irisowner/sandbox/ \
+  iris start iris && \
+  sh /home/irisowner/sandbox/install-artifact.sh zpm-0.7.1.xml \
```
Comment (isc-shuliu) [inline tests/sandbox/Dockerfile:14]: Just an idea, how about we substitute `zpm-0.7.1.xml` with an ARG that defaults to `zpm-0.7.1.xml`? It's more handy for cases where we want to install betas of 0.9.0

File: src/cls/IPM/Lifecycle/Module.cls [hunk 2]
```diff
@@ -98,17 +98,19 @@ Method %PrepareDeploy(ByRef pParams) As %Status
...
 
 			Set tSC = $Case(tExtension,
 				"GBL":##class(%IPM.Utils.Module).AddGlobalMapping(tPackageDeployNamespace, tName, tSource),
 				"INC":##class(%IPM.Utils.Module).AddRoutineMapping(tPackageDeployNamespace, tName, "INC", tSource),
 				"INT":##class(%IPM.Utils.Module).AddRoutineMapping(tPackageDeployNamespace, tName, "INT", tSource),
 				"MAC":##class(%IPM.Utils.Module).AddRoutineMapping(tPackageDeployNamespace, tName, "MAC", tSource),
 				"PKG":##class(%IPM.Utils.Module).AddPackageMapping(tPackageDeployNamespace, tName, tSource),
-				"":##class(%IPM.Utils.Module).AddRoutineMapping(tPackageDeployNamespace, tName, "ALL", tSource))
+				"":##class(%IPM.Utils.Module).AddRoutineMapping(tPackageDeployNamespace, tName, "ALL", tSource),
+				:$$$OK)
```
Comment (isc-shuliu) [inline src/cls/IPM/Lifecycle/Module.cls:112]: Question: is it our intention to allow other extension names to fail silently?

File: docker-compose.yml [hunk 3]
```diff
@@ -37,4 +36,8 @@ services:
       - IRISNAMESPACE=USER
     volumes:
       - ./:/home/irisowner/zpm/
-      - ./tests/sandbox/:/home/irisowner/sandbox/
\ No newline at end of file
+      - ./tests/sandbox/:/home/irisowner/sandbox/
+      - ./tests/sandbox/fhirpro/:/fhirpro/
```
Comment (isc-shuliu) [inline docker-compose.yml:40]: I'm not seeing a 1tests/sandbox/fhirpro/` folder. Is it supposed to be created on first run? If so, shall we add that to .gitignore?

**Review by isc-shuliu — CHANGES_REQUESTED** [review]
A few questions and comments

File: src/cls/IPM/Lifecycle/Module.cls (hunk 2 above)
Comment (isc-tleavitt) [reply src/cls/IPM/Lifecycle/Module.cls:112]: Yes.

File: docker-compose.yml (hunk 3 above)
Comment (isc-tleavitt) [reply docker-compose.yml:40]: This is debug code, should kill it.

**Review by isc-shuliu — APPROVED** [review]

**Review by isc-kiyer — APPROVED** [review]

**Pull request merged**